pkg/volume/azure_dd/vhd_util.go

```diff
@@ -56,41 +84,39 @@ func findDiskByLun(lun int, io ioHandler, exe exec.Interface) (string, error) {
 			if len(arr) < 4 {
 				continue
 			}
-			target, err := strconv.Atoi(arr[0])
+			l, err := strconv.Atoi(arr[3])
```

Comment:
the path format is `/sys/bus/scsi/devices/5:0:1:0/`
Previously i used target (i.e. 5 in the sample path) to start searching device. The new approach is to look for a device with match LUN (i.e. 1 in the sample path) and ignore target.